detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/parse/GoModuleDependencyHelper.java

```diff
@@ -34,6 +34,8 @@ public List<String> computeDependencies(String main, List<String>directs, List<S
          * requirements graph to a dependency graph.  True direct dependencies will be left unchanged.
          */
         ArrayList<String> goModGraph = new ArrayList<String>();
+        ArrayList<String> correctedDependencies = new ArrayList<String>();
+
         HashMap<String, List<String>> whyMap = whyListStructureTransform.convertWhyListToWhyMap(whyList);
         // Correct lines that get mis-interpreted as a direct dependency, given the list of direct deps, requirements graph etc.
```

Comment:
Historically we've generally used the interface (List, Map) rather than specific implementations of that interface (ArrayList, HashMap) for variable types and return values. It provide a little more flexibility.

Reply:
I have noticed that... I felt that since these were only to be used locally I could "get away with it" however I'm happy tp change them to lists...